apps/rush-lib/src/logic/buildCache/AzureStorageBuildCacheProvider.ts

```diff
@@ -119,12 +119,19 @@ export class AzureStorageBuildCacheProvider extends CloudBuildCacheProviderBase
 
     const blobClient: BlobClient = await this._getBlobClientForCacheIdAsync(cacheId);
     const blockBlobClient: BlockBlobClient = blobClient.getBlockBlobClient();
-    try {
-      await blockBlobClient.upload(entryStream, entryStream.length);
+
+    const blobAlreadyExists: boolean = await blockBlobClient.exists();
```

Comment:
Should we try securing the lease to the blob instead? Seems like that should handle concurrency better.
We should also find out what the error code is if you try to write to an immutable blob and treat that as success.

Reply:
Leasing the blob seems more complicated than it's worth. I can try to handle the simultaneous modification error.

Reply:
I'll do that in a followup